### ISSUE TITLE
Cursor missing when inserting value for new tag

### ISSUE THREAD

**Issue by kallejre** [openstreetmap/iD#7209]
TLDR: Typing cursor is missing when tags table is reordered after leaving input box for tag's key.

The new iD has a feature, that it automatically sorts tags that are inserted to "All tags" table based on predefined list (e.g source:* comes always later than roof:* or building:*). Sadly, it's too eager to sort these tags too early, when name of key is typed and cell loses focus, but value is not inserted.

Old iD editor allowed to use tab for switching between cells/input fields while adding new tags to end of table. This sometimes work in new editor as well when keys are inserted in proper order.
The most annoying exploit of the bug is when one tries to add colour=orange for a rectangular building as O is a shortcut for making circle.

**Information required by reporting guidelines**:
Bug was tested and recreated in Firefox and Opera 65.0.3467, but tested and not recreated in internet explorer 11 (there were issues showing any content in tags-table's cells). 
* Location tested: Tornimäe 7, Tallinn, Estonia. 59.43197/24.76117
* OS: Windows 10 1507 LTSB
* Browsers tested:
   * Firefox 68.0.1
   * Opera 65.0.3467
   * Internet explorer 11 (Other bug occured)
* Extensions:
   * Firefox: Multiple, most notably Adblock plus (no ads blocked on site)
   * Opera: None
* Actions taken: 
   1. Edit a (non-circular) area
   1. Add tag for something less common (source, roof or min_height)
   1. Click on plus sign (`+`) at bottom of table
   1. Type key that is considered common (height, colour). (This state is shown on screenshot).
   1. Press tab
   1. (Applies only for colour): Start typing colour, that starts with "O" (orange)
* Expected behaviour: 
   1. Cursor moves to 2nd column on the right
   1. (Applies for colour) Software suggests "Orange" as value for tag
* Actual behaviour:
   1. Table is sorted/reordered
   1. No textbox is in focus for typing
   1. If user pressed "O", area turns into circle.
* Prior art: iD v2.17.0

Best way to recall issue is by adding source tag to object and then any other key. Value cell also lost focus when clicking on it with mouse while cursor was in key cell. My suggestion is to run tags table sorting function only when entire table has lost focus. 

Screenshot: (map=20.85/59.43197/24.76117)

After that state press "Tab" ![](https://cdn.discordapp.com/attachments/662759109229281284/663007378307809287/unknown.png)

This is my first issue report on github, i hope it looks usable. I recognise IE is old and unsupported browser, but bug should be recreated on multiple environments.

**Comment by jgscherber** [comment]
Sounds similar to #1881 also

**Comment by Dimitar5555** [comment]
This issue is quite old and a lot of things have changed since (namely keys are sorted after the selection changes and there is no plus sign anymore). Candidate for closing cc: @matkoniecz
